modules/ucfunnelBidAdapter.js

```diff
@@ -0,0 +1,244 @@
+import {registerBidder} from '../src/adapters/bidderFactory';
+import {BANNER, VIDEO, NATIVE} from '../src/mediaTypes';
+
+const VER = 'ADGENT_PREBID-2018011501';
+const BIDDER_CODE = 'ucfunnel';
+
+const VIDEO_CONTEXT = {
+  INSTREAM: 0,
+  OUSTREAM: 2
+}
+
+export const spec = {
+  code: BIDDER_CODE,
+  ENDPOINT: '//hb.aralego.com/header',
+  supportedMediaTypes: [BANNER, VIDEO, NATIVE],
+  /**
+   * Check if the bid is a valid zone ID in either number or string form
+   * @param {object} bid the ucfunnel bid to validate
+   * @return boolean for whether or not a bid is valid
+   */
+  isBidRequestValid: function(bid) {
+    const isVideoMediaType = (bid.mediaTypes && bid.mediaTypes.video != null);
+    const videoContext = (bid.mediaTypes && bid.mediaTypes.video != null) ? bid.mediaTypes.video.videoContext : '';
+
+    if (typeof bid.params !== 'object' || typeof bid.params.adid != 'string') {
+      return false;
+    }
+
+    if (isVideoMediaType && videoContext === 'outstream') {
+      utils.logWarn('Warning: outstream video is not supported yet');
+      return false;
+    }
+
+    return true;
+  },
+
+  /**
+   * @param {BidRequest[]} bidRequests
+   * @param {*} bidderRequest
+   * @return {ServerRequest}
+   */
+  buildRequests: function(bids, bidderRequest) {
+    return bids.map(bid => {
+      return {
+        method: 'GET',
+        url: location.protocol + spec.ENDPOINT,
```

Comment:
As part of the 3.0 changes, requests always need to be `https` instead of being dynamic.  Can you please hard-code the protocol here (and in the related unit tests)?